calyptos/plugins/validator/topology.py

```diff
@@ -7,10 +7,25 @@ def validate(self):
         # Check each cluster
         self._single_cluster_per_host()
 
+        riakcs_keys_master = ['access-key', 'admin-email', 'admin-name', 'endpoint', 'port', 'secret-key']
         self.topology = self.environment['default_attributes']['eucalyptus']['topology']
         assert self.roles['clc']
-        assert self.roles['walrus']
         assert self.roles['user-facing']
+        if 'walrus' in self.topology and 'riakcs' in self.topology:
+            self.failure('Can not have both riakcs and walrus keys configured')
+            exit(1)
```

Comment:
@ccassler here we can still use assert in order to relay failure info. We should do one way or the other throughout the test:
https://wiki.python.org/moin/UsingAssertionsEffectively